src/optimizer.cpp

```diff
@@ -44,112 +47,181 @@ void Optimizer::getParams()
   auto getParam = utils::getParamGetter(node, name_);
   auto getParentParam = utils::getParamGetter(node, "");
 
-  getParam(model_dt_, "model_dt", 0.1);
-  getParam(time_steps_, "time_steps", 15);
-  getParam(batch_size_, "batch_size", 200);
-  getParam(iteration_count_, "iteration_count", 2);
-  getParam(temperature_, "temperature", 0.25);
-
-  getParam(base_constraints_.vx, "vx_max", 0.5);
-  getParam(base_constraints_.vy, "vy_max", 1.3);
-  getParam(base_constraints_.wz, "wz_max", 1.3);
-  constraints_ = base_constraints_;
+  auto & s = settings_;
+  getParam(s.model_dt_, "model_dt", 0.1);
+  getParam(s.time_steps_, "time_steps", 12);
+  getParam(s.batch_size_, "batch_size", 400);
+  getParam(s.iteration_count_, "iteration_count", 2);
+  getParam(s.temperature_, "temperature", 0.25);
+  getParam(s.base_constraints_.vx, "vx_max", 0.5);
+  getParam(s.base_constraints_.vy, "vy_max", 0.5);
+  getParam(s.base_constraints_.wz, "wz_max", 1.3);
+  getParam(s.sampling_std_.vx, "vx_std", 0.2);
+  getParam(s.sampling_std_.vy, "vy_std", 0.2);
+  getParam(s.sampling_std_.wz, "wz_std", 1.0);
+  s.constraints_ = s.base_constraints_;
 
-  getParam(sampling_std_.vx, "vx_std", 0.1);
-  getParam(sampling_std_.vy, "vy_std", 0.1);
-  getParam(sampling_std_.wz, "wz_std", 0.3);
   getParentParam(controller_frequency_, "controller_frequency", 0.0);
 
   std::string motion_model_name;
   getParam(motion_model_name, "motion_model", std::string("DiffDrive"));
   setMotionModel(motion_model_name);
 }
 
+// FIXME(artofnothingness) optimize this
+rcl_interfaces::msg::SetParametersResult Optimizer::dynamicParametersCallback(
+  std::vector<rclcpp::Parameter> parameters)
+{
+
```

Comment:
Need to create a mutex that this will lock to change parameters that is locked while the controller is processing data

Reply:
I think to add them without mutex firstly, just to change params while robot not moving, so params tweaking could be easier

Reply:
OK, but it eventually needs to be added, fine for just testing for now so you can change whatever, just know that there's a small potential of write/reads at the same time

Reply:
added